src/test/mon/test_mon_memory_target.cc

```diff
@@ -6,10 +6,12 @@
 #include <regex>
 #include <system_error>
 
-#include <boost/process.hpp>
+#include <boost/process/v1/io.hpp>
```

Comment:
might want add rationales in the commit message to explain why we want to use v1 namespace. this would help posterity to understand this change, and might help them to migrate to v2. i came across https://github.com/boostorg/process/issues/480 and then put my eyes on https://www.boost.org/doc/libs/1_88_0/libs/process/doc/html/index.html#version_2. both of them could help to fill other reviewers and developers with more context .